Azure/azure-cli
# [App Service] Fix#19469: `az functionapp vnet-integration add`: Add consumption plan validation

**Related command**
az functionapp vnet-integration add

**Description**
Currently, we allow virtual network integration in a function app running in a consumption plan. The changes in this PR run a validation to prevent that. This is to resolve issue #19469

**Testing Guide**
1. Create a function app with a consumption plan ( `az functionapp create -g <resource-group-name> -n <function-app-name> -s <storage-account-name> --consumption-plan-location "<location>" --functions-version 4` 
2. When attempting to add virtual network, you should see the error message "Virtual network integration is not allowed for consumption plans" (`az functionapp vnet-integration add -g <resource-group-name> -n <function-app-name> --subnet <subnet-name> --vnet <vnet-name>`)

**History Notes**

---

This checklist is used to make sure that common guidelines for a pull request are followed.
